hail/python/hail/expr/type_parsing.py

```diff
@@ -14,9 +14,10 @@
     bool = "tbool" / "bool"
     call = "tcall" / "call"
     str = "tstr" / "str"
+    nat = ("tnat" / "nat") _ "<" identifier ">"
```

Comment:
this shouldn't be an identifier, it should be an integer literal. no?

Reply:
I can add another parsing rule that explicitly only matches on integers.

Reply:
also add rule that accepts integers as nats

Reply:
actually accepting either a type or a integer literal in the ndarray last argument seems better

Reply:
(add a new rule for int literal and a visitor which returns the int)